File: mapper.js
const fs = require('fs');

const path = 'public/index.html';
let content = fs.readFileSync(path, 'utf-8');

const regex = /class="([^"]+)"/g;

content = content.replace(regex, (match, classesStr) => {
    let classes = classesStr.split(/\s+/);
    let newClasses = new Set(classes);

    const has = (cls) => newClasses.has(cls);
    const add = (cls) => newClasses.add(cls);

    if (has('bg-white') && !has('dark:bg-gray-800') && !has('dark:bg-gray-900') && !has('w-14') && !has('w-24')) {
        add('dark:bg-gray-800');
        add('transition-colors');
    }
    
    if (has('bg-gray-50') && !has('dark:bg-gray-900')) {
        add('dark:bg-gray-900');
    }

    if (has('border-gray-100') && !has('dark:border-gray-800')) {
        add('dark:border-gray-800');
    }

    if (has('border-gray-200') && !has('dark:border-gray-700')) {
        add('dark:border-gray-700');
    }

    if (has('text-gray-900') && !has('dark:text-white')) {
        add('dark:text-white');
    }

    if (has('text-gray-700') && !has('dark:text-gray-200')) {
        add('dark:text-gray-200');
    }

    if (has('text-gray-600') && !has('dark:text-gray-300')) {
        add('dark:text-gray-300');
    }

    if (has('text-gray-500') && !has('dark:text-gray-400')) {
        add('dark:text-gray-400');
    }

    if (has('text-gray-400') && !has('dark:text-gray-500')) {
        add('dark:text-gray-500');
    }

    if (has('hover:bg-gray-50')) {
        add('dark:hover:bg-gray-800');
    }

    if (has('bg-gray-200/60')) {
        add('dark:bg-gray-800/80');
    }

    return `class="${Array.from(newClasses).join(' ')}"`;
});

fs.writeFileSync(path, content, 'utf-8');
console.log('Successfully injected Dark Mode classes into index.html');

File: mapper.py
import re

def process(content):
    def repl(m):
        classesStr = m.group(1)
        classes = set(classesStr.split())
        
        if 'bg-white' in classes and 'dark:bg-gray-800' not in classes and 'dark:bg-gray-900' not in classes and 'w-24' not in classes and 'w-14' not in classes:
            classes.add('dark:bg-gray-800')
            classes.add('transition-colors')
            
        if 'bg-gray-50' in classes and 'dark:bg-gray-900' not in classes:
            classes.add('dark:bg-gray-900')
            
        if 'border-gray-100' in classes and 'dark:border-gray-800' not in classes:
            classes.add('dark:border-gray-800')
            
        if 'border-gray-200' in classes and 'dark:border-gray-700' not in classes:
            classes.add('dark:border-gray-700')
            
        if 'text-gray-900' in classes and 'dark:text-white' not in classes:
            classes.add('dark:text-white')
            
        if 'text-gray-700' in classes and 'dark:text-gray-200' not in classes:
            classes.add('dark:text-gray-200')
            
        if 'text-gray-600' in classes and 'dark:text-gray-300' not in classes:
            classes.add('dark:text-gray-300')
            
        if 'text-gray-500' in classes and 'dark:text-gray-400' not in classes:
            classes.add('dark:text-gray-400')
            
        if 'hover:bg-gray-50' in classes:
            classes.add('dark:hover:bg-gray-800')
            
        if 'bg-gray-200/60' in classes:
            classes.add('dark:bg-gray-800/80')
            
        return 'class="' + ' '.join(classes) + '"'

    return re.sub(r'class="([^"]+)"', repl, content)

with open('public/index.html', 'r', encoding='utf-8') as f:
    text = f.read()

new_text = process(text)

with open('public/index.html', 'w', encoding='utf-8') as f:
    f.write(new_text)

print("Done python mapping.")
